src/irmin-server/command.ml

```diff
@@ -544,48 +543,100 @@ struct
       end
     end
 
+    module Batch = struct
+      type batch =
+        (Store.path
+        * [ `Contents of
+            [ `Hash of Store.hash | `Value of Store.contents ]
+            * Store.metadata option
+          | `Tree of Tree.t ]
+          option)
```

Comment:
What do you think of removing this `option` in favor of adding a named variant `['Contents of ... | 'Tree of ... | 'Remove]` to clarify the `None` case?